### PR TITLE
Replace common components vuls module

### LINKED ISSUE #6222: [Remove embedded discover] Replace discover children components in vulnerabilities modules tab
## Description

Replace the `data_grid`, `search_bar`, and `doc_viewer` components for the same components placed globally in public/components/common added in https://github.com/wazuh/wazuh-dashboard-plugins/pull/6205 PR. And remove the duplicated components

Screenshot 2023-12-14 at 15 37 30

## Tasks

- [x] Replace search_bar dependencies in Vulnerabilities Detector > Dashboard tab
- [x] Replace search_bar, data_grid, doc_viewer dependencies in Vulnerabilities Detector > Inventory tab
- [x] Remove data_grid, search_bar, doc_viewers folders from vulnerabilities folder
- [x] Check that all functionalities work as expected

### PR DESCRIPTION
**Pull request by Machi3mfl**
### Description
This PR replaces and removes duplicated data grid, search bar, and doc viewer components/hooks.

Closes #6222 

## Tests

**Preconditions**
Add index `wazuh-states-vulnerabilities` with data following the [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

### Dashboard tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
|Check all the data is shown correctly (check all dashboard components are completed with data) in tops, metrics, table, etc) | :black_circle: | :black_circle: | :black_circle: |
| Click on add filter and check if the filter is applied | :black_circle: | :black_circle: | :black_circle: |
| Check dashboard table pagination works | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |
| Check dashboard responsive layout (mobile) | :black_circle: | :black_circle: | :black_circle: |

- Check all the data is shown correctly (check all dashboard components are completed with data) in tops, metrics, table, etc)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/86418a3c-5d80-4f3b-a4f8-c48b14dbe464

</details>

- Click on add filter and check if the filter is applied

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/b18742bb-ab3d-4c5b-86ec-91cbf31329e2
</details>

- Check dashboard table pagination works

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/74c7437a-969d-4596-a717-445878f00c24
</details>

- Apply filters using the search bar input

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/c80dd1dc-5fa3-4ffc-b77b-8bf7d22f0deb
</details>

- Apply filters using the search bar `add filter` feature

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/3805e4b4-0525-4ff2-84fd-48f2789a5f2e
</details>

- Check dashboard responsive layout (vertical align in mobile)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/075c0c12-e975-4de4-b1de-771b9d9648dd
</details>

### Inventory tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check all inventory table header features (sort, show/hide columns, density, etc) | :black_circle: | :black_circle: | :black_circle: |
| Check document detail flyout | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |

- Check all inventory table header features (sort, show/hide columns, density, etc)

<details>

Screen Recording 2023-12-26 at 08.54.43
</details>

- Check document detail flyout

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/f766185b-19dd-4000-865d-55d07c107b05
</details>

- Apply filters using search bar input

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/9efedc99-ec56-46ec-a0bc-971efcf3cc66
</details>

- Apply filters using the search bar `add filter` feature

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/5f037bb6-063e-4204-91f7-885040cf8a7a
</details>

### Global

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check filters persistence between tabs (inventory, dashboard) | :black_circle: | :black_circle: | :black_circle: |
| Compare results between the inventory table and dashboard table | :black_circle: | :black_circle: | :black_circle: |

- Check filters persistence between tabs (inventory, dashboard)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/5d1ed883-1b6d-4041-8092-367e6505071a
</details>

- Compare results between the inventory table and dashboard table

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/37a086ed-e6de-4154-ba5f-8a7fcaacdedf
</details>

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ All notable changes to the Wazuh app project will be documented in this file.
 
 - Support for Wazuh 4.9.0
 - Added AngularJS dependencies [#6145](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6145)
+
+### Changed
+
+- Removed embedded discover [#6120](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6120) [#6235](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6235)
 - Remove embedded discover [#6120](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6120)
```
Comment (jbiset) [inline CHANGELOG.md:15]: This line should not remain, since the change appears in the previous line

**Comment by jbiset** [comment]
:red_circle:  There is an import resolution problem in the FIM module that has become outdated with the changes in this PR

image

**Comment by jbiset** [comment]
## :red_circle:  Health check behavior
`plugins/main/public/components/health-check/services/check-pattern-support.service.ts` is different from 4.8.0 and `plugins/main/public/react-services/error-management/error-factory/errors /WarningError.ts` is probably different. This allows when the health-check passes, if there are warnings it stops and shows the continue button instead of automatically continuing with the redirection.

**_NOTE**: no se debe tener el indice creado para que aparezcan los warnings_

### Current behavior
1. The health-check is executed without debug mode.
2. The warnings appear but it quickly redirects to the overview or the corresponding callback.

### Desired behavior
1. The health-check is executed without debug mode.
3. The health-check displays the warnings and displays a continue button waiting for user interaction.

**Review by jbiset — COMMENTED** [review]
> [!NOTE]  
> I use the script on `scripts/vulnerabilities-events-injector/dataInjectScript.py` to insert vulnerability index and data index.
> I use `2.11 OSD` to test. 

## CR: :yellow_circle:  (Check comments)

## Test :green_circle: 
Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

### Dashboard tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
|Check all the data is shown correctly (check all dashboard components are completed with data) in tops, metrics, table, etc) | :green_circle: | :black_circle: | :black_circle: |
| Click on add filter and check if the filter is applied | :green_circle: | :black_circle: | :black_circle: |
| Check dashboard table pagination works | :green_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar input | :green_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :green_circle: | :black_circle: | :black_circle: |
| Check dashboard responsive layout (mobile) | :green_circle: | :black_circle: | :black_circle: |

- Check all the data is shown correctly (check all dashboard components are completed with data) in tops, metrics, table, etc) :green_circle: 

<details>

[Evidence_1.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/00c3f16a-142c-429d-8a27-b0af74ee0b87)

</details>

- Click on add filter and check if the filter is applied :green_circle: 

<details>

[Evidence_2.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/a5f84f8a-1f6c-4fb1-8491-b2d4b4e9d33b)

</details>

- Check dashboard table pagination works :green_circle: 

<details>

[Evidence_3.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/d1aec759-6306-4095-944c-d7c2f54833f8)


</details>

- Apply filters using the search bar input :green_circle: 

<details>

[Evidence_4.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/e8027a26-0f2d-4aa0-a82d-323febfd9001)


</details>

- Apply filters using the search bar `add filter` feature :green_circle: 

<details>

[Evidence_5.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/be5c606a-ea65-4405-9814-49799e456431)


</details>

- Check dashboard responsive layout (vertical align in mobile) :green_circle: 

<details>

[Evidence_6.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/59ddaf05-63c0-43c9-a8c2-2ba57a3cedc0)


</details>

### Inventory tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check all inventory table header features (sort, show/hide columns, density, etc) | :green_circle: | :black_circle: | :black_circle: |
| Check document detail flyout | :green_circle: | :black_circle: | :black_circle: |
| Apply filters using search bar input | :green_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :green_circle: | :black_circle: | :black_circle: |

- Check all inventory table header features (sort, show/hide columns, density, etc) :green_circle: 

<details>

[Evidence_7.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/38bbc61e-a709-401a-a98f-770cbb6376cb)


</details>

- Check document detail flyout :green_circle: 

<details>

[Evidence_8.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/7e18d159-5836-4fb5-b0fc-03c2fd1d924e)


</details>

- Apply filters using search bar input :green_circle: 

<details>

[Evidence_9.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/9a86f779-fe96-4a36-b034-5f4ad5379375)

</details>

- Apply filters using the search bar `add filter` feature :green_circle: 

<details>

[Evidence_10.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/8776d0b2-0941-4d74-800d-d1c015978d54)

</details>

### Global

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check filters persistence between tabs (inventory, dashboard) | :green_circle: | :black_circle: | :black_circle: |

- Check filters persistence between tabs (inventory, dashboard) :green_circle: 

<details>

[Evidence_11.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/d604534e-3117-4154-a463-21c2d5483ceb)

</details>

**Review by jbiset — APPROVED** [review]

**Comment by lucianogorza** [comment]
## Tests

### Dashboard tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
|Check all the data is shown correctly (check all dashboard components are completed with data) in tops, metrics, table, etc) | :black_circle: | :black_circle: | :black_circle: |
| Click on add filter and check if the filter is applied | :black_circle: | :black_circle: | :black_circle: |
| Check dashboard table pagination works | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |
| Check dashboard responsive layout (mobile) | :black_circle: | :black_circle: | :black_circle: |

- Check all the data is shown correctly (check all dashboard components are completed with data) in tops, metrics, table, etc)

<details>

</details>

- Click on add filter and check if the filter is applied

<details>

</details>

- Check dashboard table pagination works

<details>

</details>

- Apply filters using the search bar input

<details>

</details>

- Apply filters using the search bar `add filter` feature

<details>

</details>

- Check dashboard responsive layout (vertical align in mobile)

<details>

</details>

### Inventory tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check all inventory table header features (sort, show/hide columns, density, etc) | :black_circle: | :black_circle: | :black_circle: |
| Check document detail flyout | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |

- Check all inventory table header features (sort, show/hide columns, density, etc)

<details>

</details>

- Check document detail flyout

<details>

</details>

- Apply filters using search bar input

<details>

</details>

- Apply filters using the search bar `add filter` feature

<details>

</details>

### Global

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check filters persistence between tabs (inventory, dashboard) | :black_circle: | :black_circle: | :black_circle: |

- Check filters persistence between tabs (inventory, dashboard)

<details>

</details>

**Review by lucianogorza — APPROVED** [review]
LGMT

Look the suggestions in this [comment](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6235#issuecomment-1870386268)

**Comment by Machi3mfl** [comment]
Thanks, @lucianogorza for your suggestions.
I'd decide to apply them.
 
 
### Hide all and export

> When hide all columns and press Export formated button nothing happened. Maybe we could show a message.

The `export` button will show disable when all the columns table are hidden

Screenshot 2023-12-29 at 01 13 58

### Check document detail flyout

> The information is correct. I suggest giving more space to make reading more comfortable. Some letters are cut (like the "g" in the screenshot).

After checking the code, I found a CSS class that cut off the letters in every row. Then i removed it and the letters are shown correctly.

Screenshot 2023-12-29 at 01 26 43

**Pull request merged**